### PR TITLE
add cluster subchart

### PR DESCRIPTION
**Pull request by glitchcrab**
This PR:

- adds the upstream rook-ceph-cluster chart as a subchart

all files in `helm/rook-operator/charts/rook-ceph-cluster` are from upstream, aside from `namespace.yaml`. These are then modified to ensure they aren't deployed in the same namespace as the rest of this chart (giantswarm).

### Checklist

- [x] Update changelog in CHANGELOG.md.
- [x] Make sure `values.yaml` and `values.schema.json` are valid (more info on the values schema can be found [here](https://intranet.giantswarm.io/docs/organizational-structure/teams/halo/app-updates/helm-values-schema/)).

### CODE REVIEW

File: helm/rook-operator/charts/ceph-cluster/templates/clusterrolebinding.yaml [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+metadata:
+  name: rook-ceph-mgr-cluster-{{ .Values.global.cephClusterNamespace }}
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: ClusterRole
+  name: rook-ceph-mgr-cluster
+subjects:
+  - kind: ServiceAccount
+    name: rook-ceph-mgr
+    namespace: {{ .Values.global.cephClusterNamespace }}
```
Comment (glitchcrab) [inline helm/rook-operator/charts/ceph-cluster/templates/clusterrolebinding.yaml:15]: we cannot use `.Release.Namespace` here (as upstream does) as this chart will be installed in `giantswarm` along with the operator chart.

File: helm/rook-operator/charts/ceph-cluster/templates/namespace.yaml [hunk 2]
```diff
@@ -0,0 +1,7 @@
+{{- if ne .Release.Namespace .Values.operatorNamespace }}
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Values.global.cephClusterNamespace }}
+{{- end }}
```
Comment (glitchcrab) [inline helm/rook-operator/charts/ceph-cluster/templates/namespace.yaml:7]: we can pre-create the cluster namespace

File: helm/rook-operator/charts/ceph-cluster/templates/clusterrolebinding.yaml [hunk 3]
```diff
@@ -0,0 +1,32 @@
+{{- if .Values.global.rookEnabled }}
```
Comment (glitchcrab) [inline helm/rook-operator/charts/ceph-cluster/templates/clusterrolebinding.yaml:1]: all subchart resources are disabled unless the value is set to true in giantswarm/config

File: helm/rook-operator/charts/monitoring/templates/servicemonitor.yaml [hunk 4]
```diff
@@ -18,7 +18,7 @@ spec:
     port: http-metrics
   namespaceSelector:
     matchNames:
-    - rook-ceph
+    - {{ .Values.global.cephClusterNamespace }}
```
Comment (glitchcrab) [inline helm/rook-operator/charts/monitoring/templates/servicemonitor.yaml:21]: this was previous hard-coded which feels wrong

File: helm/rook-operator/values.yaml [hunk 5]
```diff
@@ -400,6 +400,14 @@ networkPolicy:
 # shared values with subchart.
 #
 global:
+  # namespace the Ceph components will be deployed into. Must match the
+  # namespace configured in the CephCluster CR.
+  cephClusterNamespace: "rook-ceph"
+
+  # set the ceph version to deploy. Currently only affects the toolbox
+  # pod in the ceph-cluster subchart.
+  cephVersion: ""
```
Comment (glitchcrab) [inline helm/rook-operator/values.yaml:409]: This should be set in giantswarm/config and it should match the ceph version in the `CephCluster` CR when the cluster is deployed.

**Comment by glitchcrab** [comment]
Sorry, this turned into a bigger PR than I was intending :(

I haven't deployed it yet.

File: UPDATING.md [hunk 6]
```diff
@@ -77,6 +83,37 @@ Finally, tidy up the branches we created ready for next time:
...
+
+```
+git switch --detach v1.6.0
+```
+
+- Split the subchart off into its own branch:
+
+```
+git branch -D temp-split-branch-subchart
+git subtree split -P cluster/charts/rook-ceph-cluster -b temp-split-branch-subchart
```
Comment (pipo02mix) [inline UPDATING.md:101]: 🤯

File: helm/rook-operator/charts/rook-ceph-cluster/Chart.yaml [hunk 7]
```diff
@@ -0,0 +1,7 @@
+apiVersion: v2
+description: Manages a single Ceph cluster namespace for Rook
+name: rook-ceph-cluster
+version: 0.0.1
```
Comment (pipo02mix) [inline helm/rook-operator/charts/rook-ceph-cluster/Chart.yaml:4]: is this version correct?

**Review by pipo02mix — APPROVED** [review]
nice work

File: helm/rook-operator/charts/rook-ceph-cluster/Chart.yaml (hunk 7 above)
Comment (glitchcrab) [reply helm/rook-operator/charts/rook-ceph-cluster/Chart.yaml:4]: yeah, this is just the version set in the upstream chart - i plan on sticking with it unless we need to change anything in this subchart

File: UPDATING.md (hunk 6 above)
Comment (glitchcrab) [reply UPDATING.md:101]: git definitely has some magic in it

**Comment by glitchcrab** [comment]
ok, this deploys successfully now. the rook-tools pod requires some configmaps which are created by the operator dynamically when a cluster is created so the tools pod will fail to start until a `CephCluster` CR is deployed, but this is OK.

File: UPDATING.md [hunk 8]
```diff
@@ -2,12 +2,8 @@
 
 ## Updating
 
-This process assumes that you have a master branch which is currently
-offering `v1.5.9` of `rook`, and you want to update it to `v1.6.0` from
-upstream.
-
-As this repo contains additions to the upstream chart, you will need to
-merge the changes from the new version in and assess them.
+There are two distint update processes; one for the main chart and a
```
Comment (tfussell) [inline UPDATING.md:5]:
```suggestion
There are two distinct update processes; one for the main chart and a
```

File: helm/rook-operator/charts/rook-ceph-cluster/templates/namespace.yaml [hunk 9]
```diff
@@ -0,0 +1,15 @@
+{{- if .Values.global.rookEnabled }}
+{{- if ne .Values.global.cephClusterNamespace .Values.operatorNamespace }}
+---
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: {{ .Values.global.cephClusterNamespace }}
```
Comment (tfussell) [inline helm/rook-operator/charts/rook-ceph-cluster/templates/namespace.yaml:7]: Will helm fail if this namespace already exists?

Comment (glitchcrab) [reply helm/rook-operator/charts/rook-ceph-cluster/templates/namespace.yaml]: good point - i've removed management of the namespace and i'll add an explicit instruction to the docs that this needs to be manually created.

File: helm/rook-operator/charts/ceph-cluster/templates/namespace.yaml (hunk 2 above)
Comment (glitchcrab) [reply helm/rook-operator/charts/ceph-cluster/templates/namespace.yaml]: scratch this, will document that this needs to be done manually.

**Comment by glitchcrab** [comment]
made a bunch more smaller improvements, gonna merge this in as i'm happy with it now

**Pull request merged**